### ISSUE TITLE
[Bug]: [benchmark]hnsw index, load failed after inserting 100 million data

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#25857]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230721-32827f53
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task ： fouramf-dm7sb
case : test_concurrent_locust_100m_hnsw_ddl_dql_filter_kafka_cluster

server:
```
fouramf-dm7sb-4-3814-etcd-0                                       1/1     Running            0               163m    10.104.18.26    4am-node25   <none>           <none>
fouramf-dm7sb-4-3814-etcd-1                                       1/1     Running            0               163m    10.104.17.143   4am-node23   <none>           <none>
fouramf-dm7sb-4-3814-etcd-2                                       1/1     Running            0               163m    10.104.16.146   4am-node21   <none>           <none>
fouramf-dm7sb-4-3814-kafka-0                                      1/1     Running            3 (162m ago)    163m    10.104.17.139   4am-node23   <none>           <none>
fouramf-dm7sb-4-3814-kafka-1                                      1/1     Running            2 (162m ago)    163m    10.104.18.30    4am-node25   <none>           <none>
fouramf-dm7sb-4-3814-kafka-2                                      1/1     Running            2 (162m ago)    163m    10.104.16.145   4am-node21   <none>           <none>
fouramf-dm7sb-4-3814-milvus-datacoord-59df9bfcbc-q74zt            1/1     Running            0               163m    10.104.17.134   4am-node23   <none>           <none>
fouramf-dm7sb-4-3814-milvus-datanode-79b899db4c-92nqd             1/1     Running            0               163m    10.104.18.24    4am-node25   <none>           <none>
fouramf-dm7sb-4-3814-milvus-indexcoord-6c5bbd7f7b-zsx7g           1/1     Running            0               163m    10.104.15.40    4am-node20   <none>           <none>
fouramf-dm7sb-4-3814-milvus-indexnode-55b7b99dbc-xkqj9            1/1     Running            0               163m    10.104.17.135   4am-node23   <none>           <none>
fouramf-dm7sb-4-3814-milvus-proxy-65587b79b-x2sd2                 1/1     Running            0               163m    10.104.18.23    4am-node25   <none>           <none>
fouramf-dm7sb-4-3814-milvus-querycoord-7997cdf6c9-g2hmh           1/1     Running            0               163m    10.104.15.39    4am-node20   <none>           <none>
fouramf-dm7sb-4-3814-milvus-querynode-55c687b9f-8s7jd             1/1     Running            0               163m    10.104.20.90    4am-node22   <none>           <none>
fouramf-dm7sb-4-3814-milvus-querynode-55c687b9f-cf9vd             1/1     Running            0               163m    10.104.15.41    4am-node20   <none>           <none>
fouramf-dm7sb-4-3814-milvus-rootcoord-6b988f58bf-z56nb            1/1     Running            0               163m    10.104.15.38    4am-node20   <none>           <none>
fouramf-dm7sb-4-3814-minio-0                                      1/1     Running            0               163m    10.104.17.142   4am-node23   <none>           <none>
fouramf-dm7sb-4-3814-minio-1                                      1/1     Running            0               163m    10.104.18.28    4am-node25   <none>           <none>
fouramf-dm7sb-4-3814-minio-2                                      1/1     Running            0               163m    10.104.16.144   4am-node21   <none>           <none>
fouramf-dm7sb-4-3814-minio-3                                      1/1     Running            0               163m    10.104.20.92    4am-node22   <none>           <none>
```

client log:
```
[2023-07-21 09:11:52,539 -  INFO - fouram]: [Base] Start inserting, ids: 99950000 - 99999999, data size: 100,000,000 (base.py:323)
[2023-07-21 09:11:53,887 -  INFO - fouram]: [Time] Collection.insert run in 1.3475s (api_request.py:45)
[2023-07-21 09:11:53,890 -  INFO - fouram]: [Base] Number of vectors in the collection(fouram_PlH1YXCl): 99900000 (base.py:483)
[2023-07-21 09:11:53,941 -  INFO - fouram]: [Base] Total time of insert: 2778.693s, average number of vector bars inserted per second: 35988.1426, average time to insert 50000 vectors per time: 1.3893s (base.py:394)
[2023-07-21 09:11:53,941 -  INFO - fouram]: [Base] Start flush collection fouram_PlH1YXCl (base.py:292)
[2023-07-21 09:11:56,465 -  INFO - fouram]: [Base] Params of index: [{'float_vector': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 8, 'efConstruction': 200}}}] (base.py:456)
[2023-07-21 09:11:56,465 -  INFO - fouram]: [Base] Start release collection fouram_PlH1YXCl (base.py:303)
[2023-07-21 09:11:56,470 -  INFO - fouram]: [Base] Start build index of HNSW for collection fouram_PlH1YXCl, params:{'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 8, 'efConstruction': 200}} (base.py:442)
[2023-07-21 10:37:53,026 -  INFO - fouram]: [Time] Index run in 5156.5543s (api_request.py:45)
[2023-07-21 10:37:53,027 -  INFO - fouram]: [CommonCases] RT of build index HNSW: 5156.5543s (common_cases.py:96)
[2023-07-21 10:37:53,031 -  INFO - fouram]: [Base] Params of index: [{'float_vector': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 8, 'efConstruction': 200}}}] (base.py:456)
[2023-07-21 10:37:53,031 -  INFO - fouram]: [CommonCases] Prepare index HNSW done. (common_cases.py:99)
[2023-07-21 10:37:53,031 -  INFO - fouram]: [CommonCases] No scalars need to be indexed. (common_cases.py:107)
[2023-07-21 10:37:53,034 -  INFO - fouram]: [Base] Number of vectors in the collection(fouram_PlH1YXCl): 100000000 (base.py:483)
[2023-07-21 10:37:53,035 -  INFO - fouram]: [Base] Start load collection fouram_PlH1YXCl,replica_number:1,kwargs:{} (base.py:298)
[2023-07-21 10:47:56,358 - ERROR - fouram]: RPC error: [get_loading_progress], <MilvusException: (code=52, message=deny to load, insufficient memory, please allocate more resources, collectionName: fouram_PlH1YXCl)>, <Time:{'RPC start': '2023-07-21 10:47:56.356133', 'RPC error': '2023-07-21 10:47:56.358465'}> (decorators.py:108)
[2023-07-21 10:47:56,359 - ERROR - fouram]: RPC error: [wait_for_loading_collection], <MilvusException: (code=52, message=deny to load, insufficient memory, please allocate more resources, collectionName: fouram_PlH1YXCl)>, <Time:{'RPC start': '2023-07-21 10:37:53.067509', 'RPC error': '2023-07-21 10:47:56.359837'}> (decorators.py:108)
[2023-07-21 10:47:56,360 - ERROR - fouram]: RPC error: [load_collection], <MilvusException: (code=52, message=deny to load, insufficient memory, please allocate more resources, collectionName: fouram_PlH1YXCl)>, <Time:{'RPC start': '2023-07-21 10:37:53.035240', 'RPC error': '2023-07-21 10:47:56.359967'}> (decorators.py:108)
[2023-07-21 10:47:56,361 - ERROR - fouram]: (api_response) : <MilvusException: (code=52, message=deny to load, insufficient memory, please allocate more resources, collectionName: fouram_PlH1YXCl)> (api_request.py:53)
[2023-07-21 10:47:56,361 - ERROR - fouram]: [CheckFunc] load request check failed, response:<MilvusException: (code=52, message=deny to load, insufficient memory, please allocate more resources, collectionName: fouram_PlH1YXCl)> (func_check.py:52
```

memory history : 
![image](https://github.com/milvus-io/milvus/assets/48523564/1abbda70-5e64-41f0-abbc-99c86fad7544)

deploy config : 
```
queryNode.resources.limits.cpu=4.0,queryNode.resources.limits.memory=64Gi,
queryNode.resources.requests.cpu=3.0,queryNode.resources.requests.memory=33Gi,queryNode.replicas=2,
indexNode.resources.limits.cpu=8.0,indexNode.resources.limits.memory=16Gi,
indexNode.resources.requests.cpu=5.0,indexNode.resources.requests.memory=9Gi,indexNode.replicas=1,
dataNode.resources.limits.cpu=2.0,dataNode.resources.limits.memory=4Gi,dataNode.resources.requests.cpu=2.0,
dataNode.resources.requests.memory=3Gi,dataNode.replicas=1
```

### Expected Behavior

load succeeded and searched normally. 

### Steps To Reproduce

```markdown
1. create a collection or use an existing collection
        2. build index on vector column
        3. insert a certain number of vectors
        4. flush collection
        5. build index on vector column with the same parameters
        6. build index on on scalars column or not
        7. count the total number of rows
        8. load collection
        9. perform concurrent operations
        10. clean all collections or not
```

**Comment by elstic** [comment]
@yah01 is there any other pr to merge in, or can it be verified?

**Comment by yah01** [comment]
> @yah01 is there any other pr to merge in, or can it be verified?

serveral fix patches merged, but we need to wait for the last one #25931

**Comment by elstic** [comment]
> serveral fix patches merged, but we need to wait for the last one #25931

ok

**Comment by elstic** [comment]
After verification, the problem has been fixed.
Verify the image: master-20230803-f97d5a7d
